File: src/Store/Web/Client/Authentication/PendingBankId.razor.js
export function setCookie(name, value, expiresAt) {
    let expires = "";
    if(expiresAt) {
        expires = "; expires=" + expiresAt.toUTCString();
    }
    document.cookie = name + "=" + value + expires + "; path=/; Secure; SameSite=Lax";
}

export function launchBankId(autoStartToken, returnUrl) {
    const bankIdUrl = `bankid:///?autostarttoken=${autoStartToken}&redirect=${returnUrl}`;
    window.location.href = bankIdUrl;
}

File: src/Store/Web/Client/Authentication/PendingBankId.razor.cs
using System.Runtime.InteropServices.JavaScript;

namespace Client.Authentication;

public partial class PendingBankId 
{
    [JSImport("setCookie", "PendingBankId")]
    internal static partial void SetCookie(string name, string value, [JSMarshalAs<JSType.Date>] DateTime? expiresAt = null);

    [JSImport("launchBankId", "PendingBankId")]
    internal static partial void LaunchBankId(string autoStartToken, string returnUrl);
}